articles/flask-crud-api/index.md

```diff
@@ -0,0 +1,306 @@
+In this tutorial, we will create a fully working Restful API with flask and Mysql database. Restful APIs makes it possible to establish communication between the backend and the frontend mobile and web applications, we will be using Flask-SQLAlchemy. Flask-SQLAlchemy is an extension python library for the flask web framework that adds SQLAlchemy to the flask project. SQLAlchemy is an Object Relational Mapper that provides access to SQL databases through python objects.
+
+### Prerequisites
+1. [Python](https://www.python.org/downloads/) installed on your computer.
+2. [Postman](https://www.postman.com/downloads/) installed on your computer.
+3. [Apache XAMPP](https://www.apachefriends.org/download.html) installed on your computer.
+4. Favourite IDE installed. I use [Pycharm](https://www.jetbrains.com/pycharm/download/).
+5. Basic [Python](https://www.w3schools.com/python/) knowledge. 
+
+### Project setup
+On the terminal execute the below command to create the projects' working directory and move into that directory.
+```bash
+$ mkdir flasktodo
+cd flasktodo
+```
+In the projects' working directory execute the below command to create a virtual environment for our project. Virtual environments make it easier to manage packages for various projects separately.
+```bash
+$ virtualenv venv
+```
+To activate our virtual environment, execute the below command.
+```bash
+$ source venv/Script/activate
+```
+Once the virtual environment has been activated, we need to install flask and flask-SQLAlchemy into our virtual environment using the below command.
+```bash
+(venv)$ pip install flask flask-sqlalchemy
+```
+We now need to install PyMySQL, a driver that allows for connection to the MYSQL database in python. Execute the below command to install PyMySQL.
+```bash
+(venv)$ pip install pymysql
+```
+Lastly, we need to install flask-marshmallow and marshmallow libraries which eases the work of converting Python objects to JSON and JSON to Python objects. To install flask-marshmallow and marshmallow run the below command.
```

Comment:
Consider rewording to the following to avoid repetition:

"... converting Python objects to JSON and vise versa"